### PR TITLE
[11.0][account_financial_report] Change GL PDF report layout

### PR DESCRIPTION
**Pull request by Tonow-c2c**
To fix this issue [https://github.com/OCA/account-financial-reporting/issues/556](https://github.com/OCA/account-financial-reporting/issues/556)

And this #558

### CODE REVIEW

File: account_financial_report/reports.xml [hunk 1]
```diff
@@ -17,7 +17,7 @@
       model="report_general_ledger"
       string="General Ledger"
       report_type="qweb-html"
-      name="account_financial_report.report_general_ledger_qweb"
+      name="account_financial_report.report_general_ledger_html_qweb"
```
Comment (Tonow-c2c) [inline account_financial_report/reports.xml:20]: This is for this reason [https://github.com/OCA/account-financial-reporting/issues/558](https://github.com/OCA/account-financial-reporting/issues/558)

**Review by p-tombez — COMMENTED** [review]
You may want to just remove the display of the currency, but keep the monetary widget. By doing so, you'll keep the correct rounding and thousands separator (e.g. in JPY, you don't display "cents").
Btw, tests are failling

File: account_financial_report/report/templates/general_ledger.xml [hunk 2]
```diff
@@ -325,7 +327,7 @@
                                class="o_account_financial_reports_web_action"
                                style="color: black;">
                             <!--## We don't use t-field because it throws an error on click -->
-                                <t t-esc="line.date" t-options="{'widget': 'date'}"/></a>
+                                <t t-esc="line.date" t-options="{'widget': 'date', 'format': 'dd/MM/YY'}"/></a>
```
Comment (jcoux) [inline account_financial_report/report/templates/general_ledger.xml:334]: The format date change depending of the country.
Use format date defined on language.

File: account_financial_report/report/templates/general_ledger.xml [hunk 3]
```diff
@@ -27,10 +27,9 @@
             <t t-call="account_financial_report.report_general_ledger_filters"/>
             <t t-foreach="o.account_ids" t-as="account">
                 <div class="page_break">
-                    <!-- Display account header -->
-                    <div class="act_as_table list_table" style="margin-top: 10px;"/>
-                    <div class="act_as_caption account_title"
-                         style="width: 100%">
+                  <!-- Display account header -->
+                  <div class="act_as_table list_table"/>
+                    <div class="account_title">
```
Comment (jcoux) [inline account_financial_report/report/templates/general_ledger.xml:36]: Use 4 spaces as tab like in rest of file.

File: account_financial_report/report/templates/general_ledger.xml [hunk 4]
```diff
@@ -67,9 +66,12 @@
...
                     <!-- Display account footer -->
-                    <t t-call="account_financial_report.report_general_ledger_ending_cumul">
-                        <t t-set="account_or_partner_object" t-value="account"/>
-                        <t t-set="type" t-value='"account_type"'/>
+                    <t t-if="not o.filter_partner_ids and account.account_id.user_type_id.type not in [payable, recevable] ">
+                    <!-- do not display the account ending balance on payable/recevable account type when filtering on partners -->
+                      <t t-call="account_financial_report.report_general_ledger_ending_cumul">
+                          <t t-set="account_or_partner_object" t-value="account"/>
+                          <t t-set="type" t-value='"account_type"'/>
+                      </t>
```
Comment (jcoux) [inline account_financial_report/report/templates/general_ledger.xml]: Use 4 spaces as tab like in rest of file.

File: account_financial_report/report/templates/general_ledger.xml [hunk 5]
```diff
@@ -502,15 +504,15 @@
                 <div class="act_as_cell" style="width: 2.41%;"/>
                 <!--## debit-->
                 <div class="act_as_cell amount" style="width: 6.02%;">
-                    <span t-field="account_or_partner_object.final_debit" t-options="{'widget': 'monetary', 'display_currency': res_company.currency_id}"/>
+                    <span t-field="account_or_partner_object.final_debit"  t-options="{'widget': 'float', 'precision': res_company.currency_id.decimal_places}"/>
```
Comment (jcoux) [inline account_financial_report/report/templates/general_ledger.xml]: Remove double space before `t-options`.

File: account_financial_report/report/templates/general_ledger.xml [hunk 6]
```diff
@@ -502,15 +504,15 @@
...
                 </div>
                 <!--## credit-->
                 <div class="act_as_cell amount" style="width: 6.02%;">
-                    <span t-field="account_or_partner_object.final_credit" t-options="{'widget': 'monetary', 'display_currency': res_company.currency_id}"/>
+                    <span t-field="account_or_partner_object.final_credit"  t-options="{'widget': 'float', 'precision': res_company.currency_id.decimal_places}"/>
```
Comment (jcoux) [inline account_financial_report/report/templates/general_ledger.xml]: Remove double space before `t-options`.

**Review by jcoux — CHANGES_REQUESTED** [review]

File: account_financial_report/report/templates/general_ledger.xml (hunk 2 above)
Comment (Tonow-c2c) [reply account_financial_report/report/templates/general_ledger.xml:334]: How did you that ?
@fclementic2c do you want this to ?

**Comment by Tonow-c2c** [comment]
@jcoux did you know how fix the tests ?

File: account_financial_report/report/templates/general_ledger.xml (hunk 2 above)
Comment (damdam-s) [reply account_financial_report/report/templates/general_ledger.xml:334]: you can use https://github.com/OCA/server-tools/tree/11.0/datetime_formatter#usage if adding dependency is ok

**Review by vrenaville — APPROVED** [review]

**Review by damdam-s — APPROVED** [review]

**Review by pedrobaeza — CHANGES_REQUESTED** [review]
This PR contains too much changes, and there are some polemical ones like for example the switch to landscape. I suggest you to split them in several PRs, and some of them will be merged quickly and others will require deeper discussion (like the mentioned one).

**Comment by fclementic2c** [comment]
About the "switch to landscape" for the general ledger I think we do not really have the choice:
* 1 - there are too many columns to display on this report
* 2 - landscape saves 1 third of pages with is very important for a pdf report

**Comment by fclementic2c** [comment]
About the "do not display the account ending balance on payable/recevable account type when filtering on partners"

When you want to print 1 partner to send him a copy for ex. you do not want him to see to total balance in your ledger.
I think this is a shame that an other module was created for this purpose : partner_statement

**Comment by pedroslopez** [comment]
I'm definitely on board for switching reports to landscape. As soon as we started using the module, our partners asked for this modification. It's simply too crowded to comfortably view in portrait. In any case, it could be enabled as an option. Simply switching the report's paperformat to landscape, which was my first thought, doesn't work, probably because of the conflicting names as mentioned in linked issue #558. But to be honest I'm not sure what the root cause of this is, haven't checked myself.

**Pull request closed without merging**